dask/dataframe/io/parquet/utils.py

```diff
@@ -476,3 +480,141 @@ def _scrub(i, p):
         "/".join(basepath),
         out_list,
     )  # use '/'.join() instead of _join_path to be consistent with split('/')
+
```

Comment:
Three new functions in this file are just relocated code that is now being used by all engines (previously, fastparquet was doing its own thing).